### PR TITLE
Move projection table to Technical References

### LINKED ISSUE #3346: Documentation building fails with sphinx-gallery 0.17.0
Currently building the documentation fails with:
```
generating gallery for projections/table... 

Traceback (most recent call last):
  File "/home/docs/checkouts/readthedocs.org/user_builds/pygmt-dev/conda/3312/lib/python3.12/site-packages/sphinx/events.py", line 98, in emit
    results.append(listener.handler(self.app, *args))
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/docs/checkouts/readthedocs.org/user_builds/pygmt-dev/conda/3312/lib/python3.12/site-packages/sphinx_gallery/gen_gallery.py", line 796, in generate_gallery_rst
    _replace_md5(subsection_index_path, mode="t")
  File "/home/docs/checkouts/readthedocs.org/user_builds/pygmt-dev/conda/3312/lib/python3.12/site-packages/sphinx_gallery/utils.py", line 161, in _replace_md5
    assert fname_new.endswith(".new")
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/docs/checkouts/readthedocs.org/user_builds/pygmt-dev/conda/3312/lib/python3.12/site-packages/sphinx/cmd/build.py", line 332, in build_main
    app = Sphinx(args.sourcedir, args.confdir, args.outputdir,
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/docs/checkouts/readthedocs.org/user_builds/pygmt-dev/conda/3312/lib/python3.12/site-packages/sphinx/application.py", line 292, in __init__
    self._init_builder()
  File "/home/docs/checkouts/readthedocs.org/user_builds/pygmt-dev/conda/3312/lib/python3.12/site-packages/sphinx/application.py", line 366, in _init_builder
    self.events.emit('builder-inited')
  File "/home/docs/checkouts/readthedocs.org/user_builds/pygmt-dev/conda/3312/lib/python3.12/site-packages/sphinx/events.py", line 109, in emit
    raise ExtensionError(
sphinx.errors.ExtensionError: Handler <function generate_gallery_rst at 0x7fd6af85a160> for event 'builder-inited' threw an exception (exception: )

Extension error (sphinx_gallery.gen_gallery):
Handler <function generate_gallery_rst at 0x7fd6af85a160> for event 'builder-inited' threw an exception (exception: )
```

The issue seem to be related to the [README file containing the projection table](https://github.com/GenericMappingTools/pygmt/tree/make-justify-codes-consistent/examples/projections/table). 

When removing this README file building the documentation is [partly] successfull (https://github.com/GenericMappingTools/pygmt/pull/3312/commits/a3a541637624d9d2f823a29430ce6b2e7a1f206c). I also tested changing the file extension, including `*.md` (https://github.com/GenericMappingTools/pygmt/pull/3312/commits/d355b548cec4e3c4b2040200b334817b51930f15, successfull) and `*.rst` (https://github.com/GenericMappingTools/pygmt/pull/3312/commits/9080f46a9dbfb9e5b566ceecece4f13e31694d4c, failing).

### PR DESCRIPTION
**Pull request by yvonnefroehlich**
**Description of proposed changes**

Since Sphinx-Gallery 0.17.0, building the Documentation fails (see https://github.com/GenericMappingTools/pygmt/issues/3346#issue-2421549173). It seems like the problem is the `pygmt/examples/projections/table` folder which only contains the README file with the projection table (see https://github.com/GenericMappingTools/pygmt/issues/3346#issuecomment-2250217235). To fix this and get rid of the `table` folder, this PR moves the projection table to the "Technical References".

Fixes #3346 

**Previews**
- https://pygmt-dev--3356.org.readthedocs.build/en/3356/projections/index.html
- https://pygmt-dev--3356.org.readthedocs.build/en/3356/techref/
- https://pygmt-dev--3356.org.readthedocs.build/en/3356/techref/projection_table.html

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.
- [ ] Use underscores (not hyphens) in names of Python files and directories.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash command is:

- `/format`: automatically format and lint the code

### CODE REVIEW

File: doc/techref/index.md [hunk 1]
```diff
@@ -9,4 +9,5 @@ visit the {gmt-docs}`GMT Technical Reference <reference.html>`.
 :maxdepth: 1
 
 encodings.md
+projections.md
```
[the remark this answers is not in the document]
Comment (yvonnefroehlich) [reply doc/techref/index.md]: Changed in https://github.com/GenericMappingTools/pygmt/pull/3356/commits/b351314982493e49368123d35f14ffdefdf9dc72.

I would prefer to have a separate PR for rewriting the projection table to markdown.

File: doc/techref/index.md [hunk 2]
```diff
@@ -9,4 +9,5 @@ visit the {gmt-docs}`GMT Technical Reference <reference.html>`.
 :maxdepth: 1
 
 encodings.md
+projections.rst
```
Comment (seisman) [inline doc/techref/index.md:12]: Sort by importance:
```suggestion
projections.rst
encodings.md
```

**Review by seisman — APPROVED** [review]
looks good to me except one comment

File: doc/techref/index.md (hunk 2 above)
Comment (yvonnefroehlich) [reply doc/techref/index.md]: Yes, also thought about this.

File: .github/workflows/ci_docs.yml [hunk 3]
```diff
@@ -113,7 +113,7 @@ jobs:
             sphinx-autodoc-typehints
             sphinx-copybutton
             sphinx-design
-            sphinx-gallery<0.17.0
+            sphinx-gallery
```
Comment (yvonnefroehlich) [inline .github/workflows/ci_docs.yml:116]: I am wondering if we should have two PRs:
* one for moving the projection table to fix issue #3346 
* one for unpinning the sphinx-gallery version to revert PR #3350

Comment (seisman) [reply .github/workflows/ci_docs.yml:116]: I think one PR is OK

**Pull request merged**